.github/workflows/run-tests-and-upload-coverage.yml

```
@@ -0,0 +1,103 @@
+# Copyright 2022 Democratized Data Foundation
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+name: Run Tests Workflow
+
+on:
+  pull_request:
+    branches:
+      - master
+      - develop
+
+  push:
```

todo: I think this workflow is expensive to run on every push (x2 for every push on branch flow PRs). 

```suggestion
  push:
    tags:
      - 'v[0-9]+.[0-9]+.[0-9]+'
    branches:
      - master
      - develop
```

---

I disagree with you here Shahzad - I want these tests to run, and pass, before a PR can be merged.

---

> I disagree with you here Shahzad - I want these tests to run, and pass, before a PR can be merged. I also want accurate codecov within a PR.

Where is the disagreement? Not only should these run, and pass in a PR, this should also be a required / mandatory check. Nowhere in my suggestion I suggested otherwise.

Note - I am not saying to remove this part:
```
  pull_request:
    branches:
      - master
      - develop
```

All I am saying is we shouldn't have this run on every branch, in addition to the PR, and avoid running twice on PRs opened by branch flow (i.e. dependabot PRs will run once for the PR and once for the push).

---

> Not only should these run, and pass in a PR, this should also be a required / mandatory check. 

Ah! Sorry, I thought your snippet was saying that they should only run on those branches/tags, and not in PR :) Thanks for the clarification.